Add docs for new reindex.ssl settings (#38486)
elastic/elasticsearch: docs/reference/docs/reindex.asciidoc

// TEST[s/^/PUT source\n/]
// TEST[s/otherhost:9200/\${host}/]

[float]
[[reindex-ssl]]
==== Configuring SSL parameters

Reindex from remote supports configurable SSL settings. These must be

> specified in the `elasticsearch.yml` file (or, for "secure settings",

Reviewer: ```suggestion
specified in the `elasticsearch.yml` file, with the exception of the secure settings, which you add
```

> in the elasticsearch keystore). It is not possible to configure

Reviewer: ```suggestion
in the Elasticsearch keystore. It is not possible to configure
```

`reindex.ssl.truststore.path`.

`reindex.ssl.truststore.password`::
The password to the truststore (`reindex.ssl.truststore.path`).
This setting cannot be used with `reindex.ssl.truststore.secure_password`.

`reindex.ssl.truststore.secure_password` (<<secure-settings,Secure>>)::
The password to the truststore (`reindex.ssl.truststore.path`).
This setting cannot be used with `reindex.ssl.truststore.password`.

`reindex.ssl.truststore.type`::
The type of the truststore (`reindex.ssl.truststore.path`).
Must be either `jks` or `PKCS12`. If the truststore path ends in ".p12", ".pfx"
or "pkcs12", this setting defaults to `PKCS12`. Otherwise, it defaults to `jks`.

`reindex.ssl.verification_mode`::
Indicates the type of verification to protect against man in the middle attacks
and certificate forgery. 
One of `full` (verify the hostname and the certificate path), `certificate`
(verify the certificate path, but not the hostname) or `none` (perform no
verification - this is strongly discouraged in production environments).
Defaults to `full`.

`reindex.ssl.certificate`::
Specifies the path to the PEM encoded certificate (or certificate chain) to be

> used for http client authentication (if required by the remote cluster)

Reviewer: ```suggestion
used for HTTP client authentication (if required by the remote cluster).
```

This setting requires that `reindex.ssl.key` also be set.
You cannot specify both `reindex.ssl.certificate` and `reindex.ssl.keystore.path`.

`reindex.ssl.key`::
Specifies the path to the PEM encoded private key associated with the
certificate used for client authentication (`reindex.ssl.certificate`).
You cannot specify both `reindex.ssl.key` and `reindex.ssl.keystore.path`.

`reindex.ssl.key_passphrase`::
Specifies the passphrase to decrypt the PEM encoded private key
(`reindex.ssl.key`) if it is encrypted.
Cannot be used with `reindex.ssl.secure_key_passphrase`. 

`reindex.ssl.secure_key_passphrase` (<<secure-settings,Secure>>)::
Specifies the passphrase to decrypt the PEM encoded private key
(`reindex.ssl.key`) if it is encrypted.
Cannot be used with `reindex.ssl.key_passphrase`. 

`reindex.ssl.keystore.path`::
Specifies the path to the keystore that contains a private key and certificate

> to be used for http client authentication (if required by the remote cluster).

Reviewer: ```suggestion
to be used for HTTP client authentication (if required by the remote cluster).
```
